pycc/ccresponse.py

```diff
@@ -0,0 +1,361 @@
+        r_X1 += contract('imef,amef->ia', X2, (2.0*hbar.Hvovv - hbar.Hvovv.swapaxes(2,3)))
+        r_X1 -= contract('mnae,mnie->ia', X2, (2.0*hbar.Hooov - hbar.Hooov.swapaxes(0,1)))
+
+        return r_X1
+
+    def r_X2(self, pertbar, omega):
+        contract = self.contract
+        o = self.ccwfn.o
+        v = self.ccwfn.v
+        X1 = self.X1
+        X2 = self.X2
+        t2 = self.ccwfn.t2
+        hbar = self.hbar
+        L = self.H.L
+
+        Zvv = contract('amef,mf->ae', (2.0*hbar.Hvovv - hbar.Hvovv.swapaxes(2,3)), X1)
+        Zvv -= contract('mnef,mnaf->ae', L[o,o,v,v], X2)
+
+        Zoo = -1.0*contract('mnie,ne->mi', (2.0*hbar.Hooov - hbar.Hooov.swapaxes(0,1)), X1)
+        Zoo -= contract('mnef,inef->mi', L[o,o,v,v], X2)
+
+        r_X2 = pertbar.Avvoo - 0.5 * omega*X2
+        r_X2 += contract('ie,abej->ijab', X1, hbar.Hvvvo)
+        r_X2 -= contract('ma,mbij->ijab', X1, hbar.Hovoo)
+        r_X2 += contract('mi,mjab->ijab', Zoo, t2)
+        r_X2 += contract('ae,ijeb->ijab', Zvv, t2)
+        r_X2 += contract('ijeb,ae->ijab', X2, hbar.Hvv)
+        r_X2 -= contract('mjab,mi->ijab', X2, hbar.Hoo)
+        r_X2 += 0.5*contract('mnab,mnij->ijab', X2, hbar.Hoooo)
+        r_X2 += 0.5*contract('ijef,abef->ijab', X2, hbar.Hvvvv)
+        r_X2 -= contract('imeb,maje->ijab', X2, hbar.Hovov)
+        r_X2 -= contract('imea,mbej->ijab', X2, hbar.Hovvo)
+        r_X2 += 2.0*contract('miea,mbej->ijab', X2, hbar.Hovvo)
+        r_X2 -= contract('miea,mbje->ijab', X2, hbar.Hovov)
+
+        r_X2 = r_X2 + r_X2.swapaxes(0,1).swapaxes(2,3)
+
+        return r_X2
+
+    def pseudoresponse(self, pertbar, X1, X2):
+        contract = self.ccwfn.contract
+        polar1 = 2.0 * contract('ai,ia->', np.conj(pertbar.Avo), X1)
+        polar2 = 2.0 * contract('ijab,ijab->', np.conj(pertbar.Avvoo), (2.0*X2 - X2.swapaxes(2,3)))
+
+        return -2.0*(polar1 + polar2)
+        
+
+class pertbar(object):
+    def __init__(self, pert, ccwfn):
+        o = ccwfn.o
+        v = ccwfn.v
+        t1 = ccwfn.t1
+        t2 = ccwfn.t2
+        contract = opt_einsum.contract
+
+        self.Aov = pert[o,v].copy()
+
+        self.Aoo = pert[o,o].copy()
+        self.Aoo += contract('ie,me->mi', t1, pert[o,v])
+
+        self.Avv = pert[v,v].copy()
+        self.Avv -= contract('ma,me->ae', t1, pert[o,v])
+
+        self.Avo = pert[v,o].copy()
+        self.Avo += contract('ie,ae->ai', t1, pert[v,v])
+        self.Avo -= contract('ma,mi->ai', t1, pert[o,o])
+        self.Avo += contract('miea,me->ai', (2.0*t2 - t2.swapaxes(2,3)), pert[o,v])
+        self.Avo -= contract('ie,ma,me->ai', t1, t1, pert[o,v])
+
+        self.Aovoo = contract('ijeb,me->mbij', t2, pert[o,v])
+
+        self.Avvvo = -1.0*contract('miab,me->abei', t2, pert[o,v])
+  
+        # This is intended to match ugacc, but I need to re-derive all of these expressions for consistency
+        # Also note the incorrectness of my tensor notation for Avvoo
```

Comment:
would `sed -i '' 's/Avvoo/Aoovv` ccresponse.py` be sufficient, or is there a reason (besides matching UGACC) that you've kept the incorrect notation?